draft-irtf-cfrg-opaque.md

```diff
@@ -2264,6 +2264,23 @@ attacker from using this data to mount a server spoofing attack. Supporting impl
 need to consider allowing separate AKE and OPRF algorithms in cases where the HSM is
 incompatible with the OPRF algorithm.
 
+## Client Authentication Using Credentials
+
+For scenarios in which the client has access to private state that can be persisted across
+registration and login, the client can back up the `randomized_password` variable (as
+computed in {{finalize-request}}) so that upon a future login attempt, the client can
+authenticate to the server using `randomized_password` instead of the original password.
+This can be achieved by supplying an arbitrary password as input to
+`CreateCredentialRequest` in the login phase, and then instantiating the `randomized_password`
+variable in `RecoverCredentials` (invoked by `GenerateKE3`) rather than computing it from
```

Comment:
```suggestion
This can be achieved by supplying an arbitrary password as input to
`CreateCredentialRequest` in the login phase, and then using the `randomized_password` from
the backup in `RecoverCredentials` (invoked by `GenerateKE3`) rather than computing it from
```

Reply:
(Updated)